examples/no-prefix/main.tf

```diff
@@ -0,0 +1,56 @@
+##############################################################################
+# Resource Group
+# (if var.resource_group is null, create a new RG using var.prefix)
+##############################################################################
+
+resource "ibm_resource_group" "resource_group" {
+  count    = var.resource_group != null ? 0 : 1
+  name     = "${var.prefix}-rg"
+  quota_id = null
+}
+
+data "ibm_resource_group" "existing_resource_group" {
+  count = var.resource_group != null ? 1 : 0
+  name  = var.resource_group
+}
+
+#############################################################################
+# Provision cloud object storage and bucket
+#############################################################################
+
+resource "ibm_resource_instance" "cos_instance" {
+  count             = var.enable_vpc_flow_logs ? 1 : 0
+  name              = "${var.prefix}-vpc-logs-cos"
+  resource_group_id = var.resource_group != null ? data.ibm_resource_group.existing_resource_group[0].id : ibm_resource_group.resource_group[0].id
+  service           = "cloud-object-storage"
+  plan              = var.cos_plan
+  location          = var.cos_location
+}
+
+resource "ibm_cos_bucket" "cos_bucket" {
+  count                = var.enable_vpc_flow_logs ? 1 : 0
+  bucket_name          = "${var.prefix}-vpc-logs-cos-bucket"
+  resource_instance_id = ibm_resource_instance.cos_instance[0].id
+  region_location      = var.region
+  storage_class        = "standard"
+}
+
+#############################################################################
+# Provision VPC
+#############################################################################
+
+module "slz_vpc" {
+  source                                 = "../../"
+  resource_group_id                      = var.resource_group != null ? data.ibm_resource_group.existing_resource_group[0].id : ibm_resource_group.resource_group[0].id
+  region                                 = var.region
+  name                                   = var.name
```

Comment:
don't we now want to update this to `name = var.prefix` so our tests will pass a unique prefix to it?

Reply:
ah actually never mind, we are doing this directly in the test

Reply:
The prefix should be null. We want to pass a unique name to it.

Reply:
@ocofaigh in the pr_test code there is a different unique value being generated for the name, and it is passed into the name variable for the test and checked afterwards. The prefix and name will use different and random/unique values.

Done this way because the unit test doesn't have direct access to the unique value used as part of the prefix, and the entire prefix is already a bit long.